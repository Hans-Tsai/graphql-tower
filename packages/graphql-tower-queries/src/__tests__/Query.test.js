import faker from 'faker';
import { graphql, GraphQLSchema, GraphQLObjectType, GraphQLInt } from 'graphql';
import { Query } from '../';

describe('Query', () => {
  it('snapshot', async () => {
    const query = new Query();
    expect(query).toMatchSnapshot();
  });

  it('successfully query', async () => {
    const value = faker.random.number();
    const resolve = jest.fn(() => Promise.resolve(value));

    const Node = class extends Query {
      type = GraphQLInt;
      resolve = resolve;
    };


    const schema = new GraphQLSchema({
      query: new GraphQLObjectType({ name: 'Query', fields: { node: new Node() } }),
    });

    const { data } = await graphql(schema, 'query { node }');
    expect(data).toEqual({ node: value });
    expect(resolve).toHaveBeenCalledTimes(1);
  });
});
