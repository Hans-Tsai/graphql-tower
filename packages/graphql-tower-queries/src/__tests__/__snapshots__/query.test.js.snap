// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query Query 1`] = `
QueryNode {
  "_": Object {
    "afterware": Array [],
    "middleware": Array [],
    "parentArgs": undefined,
    "resolve": [Function],
  },
  "args": Object {},
  "resolve": [Function],
}
`;
